core/src/test/scala/unit/kafka/server/AlterUserScramCredentialsRequestTest.scala

```diff
@@ -43,20 +41,27 @@ import scala.jdk.CollectionConverters._
  * Also tests the Alter and Describe APIs for the case where credentials are successfully altered/described.
  */
 class AlterUserScramCredentialsRequestTest extends BaseRequestTest {
+
+  private[this] var className = classOf[AlterCredentialsTest.TestPrincipalBuilderReturningAuthorized].getName
+
+  override def setUp(): Unit = {
+    // do nothing as we will setup cluster by 'before'
+  }
+
+  @BeforeEach
+  def before(info: TestInfo): Unit = {
+    if (info.getDisplayName.contains("NotAuthorized"))
```

Comment:
This is pretty brittle. Why do we need it?

Reply:
It required to override the properties of starting server according to method name. You are right, current code is weird so I will separate the tests which need different props to another file.